src/components/drawer/drawer.component.js

```diff
@@ -21,20 +22,63 @@ const Drawer = ({
   animationDuration,
   ...props
 }) => {
+  const drawerSidebarContentRef = useRef();
   const isControlled = expanded !== undefined;
   const [isExpandedInternal, setIsExpandedInternal] = useState(defaultExpanded || false);
-  const [isFirstOpen, setFirstOpen] = useState(true);
+  const [isOpening, setIsOpening] = useState(false);
+  const [isClosing, setIsClosing] = useState(false);
   const isExpanded = isControlled ? expanded : isExpandedInternal;
+
+  const duration = useCallback(() => {
+    if (typeof animationDuration === 'string' && animationDuration.indexOf('ms') !== -1) {
+      const animationTime = animationDuration.substring(0, animationDuration.length - 2);
+      return animationTime;
+    }
+
+    if (typeof animationDuration === 'string' && animationDuration.indexOf('.') !== -1) {
+      const animationTime = animationDuration.substring(0, animationDuration.length - 1);
+      return parseFloat(animationTime) * 1000;
+    }
+
+    return animationDuration;
+  }, [animationDuration]);
+
+  const toggling = useCallback(() => {
+    const timeout = duration();
+    if (!isExpanded) {
+      setIsOpening(true);
+      setTimeout(() => {
+        setIsOpening(false);
+      }, timeout);
+    } else {
+      setIsClosing(true);
+      setTimeout(() => {
+        setIsClosing(false);
+      }, timeout);
+    }
+  }, [duration, isExpanded]);
+
   const toggleDrawer = useCallback((ev) => {
     if (!isControlled) setIsExpandedInternal(!isExpanded);
     if (onChange) onChange(ev, !isExpanded);
-    setFirstOpen(false);
-  }, [isControlled, isExpanded, onChange]);
+    if (isExpanded) {
+      drawerSidebarContentRef.current.scrollTop = 0;
+    }
+
+    toggling();
+  }, [toggling, isControlled, isExpanded, onChange]);
 
-  const closedClass = isFirstOpen ? '' : 'closed';
   const guid = useRef(createGuid());
   const sidebarId = `DrawerSidebar_${guid.current}`;
 
+  const classes = useCallback(() => {
```

Comment:
it's a method so the name should be a verb to avoid confusion

Reply:
renamed to `getClassNames`